### ISSUE TITLE
Is this code your experiments code in your paper?

### ISSUE THREAD

**Issue by KaichuangZhang** [halquabeh/ACML-2023-FPOGD-Code#1]

**Comment by halquabeh** [comment]
Thanks for your comment. yes, partially, the code can regenerate the linear algorithms ressults in the paper, let me know if you need further help.

**Comment by KaichuangZhang** [comment]
> Thanks for your comment. yes, partially, the code can regenerate the linear algorithms ressults in the paper, let me know if you need further help.

Ok. Thank you for your response. That means the RFF in algorithm 1  is not implemented in the code, right?

**Comment by halquabeh** [comment]
exactly, the RFF should be added if you consider kernelzation of the input space, I Will update the code shortly once the RFF is polished.

**Comment by KaichuangZhang** [comment]
> exactly, the RFF should be added if you consider kernelzation of the input space, I Will update the code shortly once the RFF is polished.

OK. I understand. Thanks. Nice work.

**Comment by halquabeh** [comment]
Thanks alot :)

**Comment by KaichuangZhang** [comment]
Hello, Halquabeh. I am trying to add RFF.  What I do is transform the sample into another space. And, other parts are the same as the linear algorithm. right?

**Comment by halquabeh** [comment]
Hi, basically, few steps, 1- Generate the probability that we would sample from (equation 13 in the paper), such probability depends on the kernel used, in my experiments we used Gaussian, and the probability is Normal with zero mean and covariance equals to the kernel width (hyper-parameter) 2- Given the probability we can now map any new example (online) to the new space (R^D, where D is the number of random features), check the  last line before th8. In practice, we generate a random Matrix with standard deviation and scale it with the width (i.e. if we consider Matlab language: W = sqrt(sigma) * randn(d,D) where d is original dimension of data) and  then map x to new space as follow 
x1 = cos(W'*x);
x2 = sin(W'*x);
new_x = 1/sqrt(D)*[x1;x2];
I wish this helped.

**Comment by KaichuangZhang** [comment]
> Hi, basically, few steps, 1- Generate the probability that we would sample from (equation 13 in the paper), such probability depends on the kernel used, in my experiments we used Gaussian, and the probability is Normal with zero mean and covariance equals to the kernel width (hyper-parameter) 2- Given the probability we can now map any new example (online) to the new space (R^D, where D is the number of random features), check the last line before th8. In practice, we generate a random Matrix with standard deviation and scale it with the width (i.e. if we consider Matlab language: W = sqrt(sigma) * randn(d,D) where d is original dimension of data) and then map x to new space as follow x1 = cos(W'*x); x2 = sin(W'_x); new_x = 1/sqrt(D)_[x1;x2]; I wish this helped.

Thanks.
